### ISSUE TITLE
[BUG] Editor does not resize when focus switched from TextFeild widget to Editor

### ISSUE THREAD

**Issue by devanshu89w** [tneotia/html-editor-enhanced#122]
**Editor fails to resize**
When typing first in a TextField and moving on to HtmlEditor, the Editor does not resizes.

**To Reproduce**
1. Create a sample app, and paste the below snippet as Scafold's body

https://gist.github.com/devanshu89w/6e69f1b34b6be669c4cedd5c3a32c6ab

2. Click on 'Field 2'
3. We can see the editor resized it's height when Keyboard was shown.
3. Restart the app, and click on 'Field 1' and then click on 'Field 2'
4. The HTML editor did not resize.

**Expected behavior**
The HTML editor should resize after step 3

**Additional context**
I have tried adding a focus node to TextField and on onFocus callback of HTML Editor unfocused field 1 node. This caused another issue wherein to type in editor I had to click/touch twice. then the keyboard shows up.

**Comment by devanshu89w** [comment]
@tneotia , I did some initial debugging and the issue seems to be that visibility has changed before the WebView window was in focus. due to this, visibility change did add the new visibleFraction value in the stream, but on window focus the 

`visibleDecimal = await visibleStream.stream.first;`

this waits the value till another visibility change happens and a value is added in the stream again.

**Comment by tneotia** [comment]
Ah I see, this should be easily solved by listening for visibility changes within `initState` rather than waiting until window focus. Thanks for your report! I'll try to have this fixed soon but I have been busy with school and other projects recently.

**Comment by devanshu89w** [comment]
@tneotia, I have tried this, but it looks to get in loop, where we listen for visibility changes and then we change  the height of the document which in turn changes the visibility of the widget again.

**Comment by tneotia** [comment]
Hmmm okay - does this happen even without the restart? Like if you load the app and then click in field 1, type some, click in field 2, does it resize then? Or does it require hot restart?

**Comment by tneotia** [comment]
Sorry for the delay, but I fixed this issue. Basically I cached the value of the visibility widget because it gives a value even when the widget is not focused, then in the onFocused callback of the webview I check the cached value and update the height.

**Comment by devanshu89w** [comment]
@tneotia, thanks for the support. I made a similar fix in local copy earlier to work past the issue. however, I would like to roll back them and use the out of the box fix in plugin. do you plan to release the fix soon. if not can you share the snippet, I will compare and adjust my changes.

Thanks again for your support.

**Comment by tneotia** [comment]
It is released already in 2.3.0, you can update your dependency and it should work.
